test/e2e/kubectl/debug.go

```diff
@@ -0,0 +1,188 @@
+	testutils "k8s.io/kubernetes/test/utils"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+var _ = SIGDescribe("kubectl debug", func() {
+	defer ginkgo.GinkgoRecover()
+	f := framework.NewDefaultFramework("kubectl-debug")
+	f.NamespacePodSecurityLevel = admissionapi.LevelPrivileged
+
+	var ns string
+	var c clientset.Interface
+	ginkgo.BeforeEach(func() {
+		c = f.ClientSet
+		ns = f.Namespace.Name
+	})
+
+	ginkgo.Describe("custom profile", func() {
+		ginkgo.It("should be applied on static profiles on ephemeral container", func(ctx context.Context) {
+			debugPodName := "e2e-test-debug-custom-profile-pod-ephemeral"
+			customProfileImage := imageutils.GetE2EImage(imageutils.BusyBox)
+			ginkgo.By("running the image " + customProfileImage)
+			e2ekubectl.RunKubectlOrDie(ns,
+				"run", debugPodName,
+				"--image="+customProfileImage,
+				podRunningTimeoutArg,
+				"--labels=run="+debugPodName,
+				"--", "sh", "-c",
+				"sleep 3600")
+
+			ginkgo.By("verifying the pod " + debugPodName + " is running")
+			label := labels.SelectorFromSet(map[string]string{"run": debugPodName})
+			err := testutils.WaitForPodsWithLabelRunning(c, ns, label)
+			if err != nil {
+				framework.Failf("Failed getting pod %s: %v", debugPodName, err)
+			}
+
+			tmpDir, err := os.MkdirTemp("", "test-custom-profile-debug")
+			customProfileFile := "custom.yaml"
+			framework.ExpectNoError(err)
+			defer os.Remove(tmpDir) //nolint:errcheck
+			framework.ExpectNoError(os.WriteFile(filepath.Join(tmpDir, customProfileFile), []byte(`
+securityContext:
+  capabilities:
+    add:
+      - NET_ADMIN
+env:
+  - name: REQUIRED_ENV_VAR
+    value: value2
+`), os.FileMode(0755)))
+			ginkgo.By("verifying ephemeral container has correct fields")
+			e2ekubectl.RunKubectlOrDie(ns,
+				"debug",
+				debugPodName,
+				"--image="+imageutils.GetE2EImage(imageutils.Nginx),
+				"--container=debugger",
+				"--profile=general", // general profile sets capability to SYS_PTRACE and custom should overwrite it to NET_ADMIN
+				"--custom="+filepath.Join(tmpDir, customProfileFile),
+				"--", "sh", "-c",
+				"sleep 3600")
+
+			ginkgo.By("verifying the container debugger is running")
+			err = wait.PollUntilContextTimeout(ctx, 3*time.Second, 2*time.Minute, false, func(ctx context.Context) (bool, error) {
+				running := e2ekubectl.RunKubectlOrDie(ns, "get", "pod", debugPodName, "-o", "template", `--template={{if (exists . "status" "ephemeralContainerStatuses")}}{{range .status.ephemeralContainerStatuses}}{{if (and (eq .name "debugger") (exists . "state" "running"))}}true{{end}}{{end}}{{end}}`)
+				if running != "true" {
+					return false, nil
+				}
+				return true, nil
+			})
+			framework.ExpectNoError(err)
+			output := e2ekubectl.RunKubectlOrDie(ns, "get", "pod", debugPodName, "-o", "jsonpath={.spec.ephemeralContainers[*]}")
+			ginkgo.By("verifying NET_ADMIN is added")
+			if !strings.Contains(output, "NET_ADMIN") {
```

Comment:
Change all three conditions to use this form:
```
gomega.Expect(output).To(gomega.MatchError(gomega.ContainSubstring("NET_ADMIN")))
```
which on error will provide more explainable output, see [here for reference](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/writing-good-e2e-tests.md#debuggability).

Reply:
Used in this form;

```go
gomega.Expect(output).To(gomega.ContainSubstring("NET_ADMIN"))
```